components/eam/src/chemistry/utils/tracer_data.F90

```diff
@@ -2431,7 +2592,83 @@ subroutine vert_interp( ncol, levsiz, pin, pmid, datain, dataout )
 
 
   end subroutine vert_interp
-
+!------------------------------------------------------------------------------
+  SUBROUTINE vert_interp_uci(ncol,levsiz,pin,pint,datain,dataout)
+!-------------------------------------------------------------------------- 
+    ! 
+    ! Interpolate data from current time-interpolated values to model levels
+    !--------------------------------------------------------------------------
+    implicit none
+    ! Arguments
+    !
+    integer,  intent(in)  :: ncol
+    integer,  intent(in)  :: levsiz
+    real(r8), intent(in)  :: pin(levsiz+1) !inputdata interface level pressure
+    real(r8), intent(in)  :: pint(pcols,pver+1) !model interface level pressures
+    real(r8), intent(in)  :: datain(pcols,levsiz)
+    real(r8), intent(out) :: dataout(pcols,pver)
+    !
+    ! local storage
+    !
+    integer ::  i,k                   ! longitude index
+    ! Initialize index array
+    !
+    do i=1,ncol
+        do k=1,pver
+        call vert_interp_uci_single(pint(i,k+1),pint(i,k),dataout(i,k),pin,datain(i,:),levsiz)
+        enddo
+    enddo
```

Comment:
Fortran uses column-major indexing, i.e., the fastest varying index is always the innermost. The innermost index should be the innermost loop in a nested loop for better performance. I recommend you swap the order of the two loops.

Reply:
Hi Qi, it is updated as suggested.